Backend/src/routes/api.js

```diff
@@ -344,6 +211,23 @@ function openSocket() {
     if (solarCarData.solar_car_connection.length > 0) {
       solarCarData.solar_car_connection[0] = false;
       frontendData.solar_car_connection[0] = false;
+      // If recording, replace the latest solar_car_connection value in file with false
+      if(doRecord) {
+        fs.open("recordedData/sessions/" + currentSession + ".bin", "r+", (err, fd) => {
+          if(!err) {
+            fs.write(
+                fd, Buffer.alloc(1, false), 0, 1, fs.fstatSync(fd).size - 1,
+                (err, bw, buf) => {
+                  if(!err) {
+                    // Successfully wrote byte to offset
+                  } else {
+                    console.warn("ERROR: Error writing to file when lost connection");
+                  }
+                }
+            );
+          }
+        });
+      }
```

Comment:
Lets use a if(err) exit kind of structure here. It's hard to follow. Instead of having the error condition at the bottom you should have it at the top

Reply:
Addressed